patroni/ctl.py

```diff
@@ -1372,3 +1373,246 @@ def format_pg_version(version):
+
+    no_value_msg = '#FIXME'
+    pg_version = None
+
+    dynamic_config = Config.get_default_config()
+    dynamic_config['postgresql']['parameters'] = dict(dynamic_config['postgresql']['parameters'])
+    config = {
+        'scope': scope or no_value_msg,
+        'name': socket.gethostname(),
+        'bootstrap': {
+            'dcs': dynamic_config
+        },
+        'postgresql': {
+            'parameters': None,
+            'connect_address': no_value_msg,
+            'listen': no_value_msg,
+        },
+    }
+    # for a running instance bin_dir will be gathered from the postmaster process
+    if sample and bin_dir:
+        config['postgresql']['bin_dir'] = bin_dir
+
+    if sample:  # some sane defaults or values set via Patroni env vars
+        replicator = os.getenv('PATRONI_REPLICATION_USERNAME', 'replicator')
+        config['postgresql']['authentication'] = {
+            'superuser': {
+                'username': os.getenv('PATRONI_SUPERUSER_USERNAME', 'postgres'),
+                'password': os.getenv('PATRONI_SUPERUSER_PASSWORD', no_value_msg)
+            },
+            'replication': {
+                'username': replicator,
+                'password': os.getenv('PATRONI_REPLICATION_PASSWORD', no_value_msg)
+            }
+        }
+        config['postgresql']['pg_hba'] = [
+            'host all all 0.0.0.0/0 md5',
+            f'host replication {replicator} all md5'
+        ]
+    else:  # or real values
+        enrich_config_from_running_instance(config, no_value_msg, dsn)
+
+    if 'bin_dir' in config['postgresql']:
+        # obtain version from the binary
+        try:
+            pg_version = postgres_major_version_to_int(get_major_version(config['postgresql']['bin_dir'] or None))
+        except PatroniException as e:
+            raise PatroniCtlException(str(e))
+
+        # add version-specific configuration
+        if dynamic_config['postgresql']['parameters'].keys().isdisjoint({'wal_keep_size', 'wal_keep_segments'}):
+            from patroni.postgresql.config import ConfigHandler
+            wal_keep_param = 'wal_keep_segments' if pg_version < 130000 else 'wal_keep_size'
+            config['bootstrap']['dcs']['postgresql']['parameters'][wal_keep_param] =\
+                ConfigHandler.CMDLINE_OPTIONS[wal_keep_param][0]
+        if sample and pg_version >= 110000:
+            config['bootstrap']['dcs']['postgresql']['use_pg_rewind'] = True
+            config['postgresql']['authentication']['rewind'] = {
+                'username': os.getenv('PATRONI_REWIND_USERNAME', 'rewind_user'),
+                'password': no_value_msg
+            }
+    else:
+        config['postgresql']['bin_dir'] = ''
+
+    # redundant values from the default config
+    del config['bootstrap']['dcs']['postgresql']['parameters']['listen_addresses']
+    del config['bootstrap']['dcs']['postgresql']['parameters']['port']
+    del config['bootstrap']['dcs']['postgresql']['parameters']['cluster_name']
```

Comment:
Lets avoid putting `listen_addresses`, `port`, and `cluster_name` to the `config['bootstrap']['dcs']['postgresql']` in the first place.

First two are only needed for `config['postgresql']['listen'] = f'{listen_addresses}:{port}'` and `cluster_name` we don't need at all.